unittests/CppInterOp/DynamicLibraryManagerTest.cpp

```diff
@@ -54,3 +59,30 @@ TEST(DynamicLibraryManagerTest, Sanity) {
   // invalidated...
   // EXPECT_FALSE(Cpp::GetFunctionAddress("ret_zero"));
 }
+
+TEST(UtilsPlatform, DLTest) {
+  std::string err = "";
+  // CPPINTEROP_LIB_TestSharedLib_DIR_PREFIX specified by cmake though target
+  // definitions
+#if defined(__APPLE__)
+  auto dlopen_handle = Cpp::utils::platform::DLOpen(
+      CPP_INTEROP_STRINGIFY(
+          CPPINTEROP_LIB_TestSharedLib_DIR_PREFIX) "./libTestSharedLib.dylib",
+      err);
+#elif defined(_WIN32)
+  auto dlopen_handle = Cpp::utils::platform::DLOpen(
+      CPP_INTEROP_STRINGIFY(
+          CPPINTEROP_LIB_TestSharedLib_DIR_PREFIX) "./libTestSharedLib.dll",
+      err);
+#else
+  auto dlopen_handle = Cpp::utils::platform::DLOpen(
+      CPP_INTEROP_STRINGIFY(
+          CPPINTEROP_LIB_TestSharedLib_DIR_PREFIX) "./libTestSharedLib.so",
+      err);
+#endif
+  EXPECT_TRUE(dlopen_handle);
+  EXPECT_TRUE(err.empty());
+  Cpp::utils::platform::DLOpen("missing", err);
+  EXPECT_TRUE(err.find("no such file") != std::string::npos ||
+              err.find("No such file") != std::string::npos);
```

Comment:
```suggestion
  EXPECT_FALSE(err.empty());
```

Reply:
I feel `no such file` is more verbose and the exact error we expect.